components/Ink/src/private/MDCLegacyInkLayer.m

```diff
@@ -621,11 +654,18 @@ - (void)animationDidStop:(__unused CAAnimation *)anim
   [shapeLayer removeFromSuperlayer];
   [shapeLayer removeAllAnimations];
 
-  if ([shapeLayer isMemberOfClass:[MDCLegacyInkLayerForegroundRipple class]]) {
+  if ([shapeLayer isKindOfClass:[MDCLegacyInkLayerForegroundRipple class]]) {
     [self.foregroundRipples removeObject:(MDCLegacyInkLayerForegroundRipple *)shapeLayer];
-  } else if ([shapeLayer isMemberOfClass:[MDCLegacyInkLayerBackgroundRipple class]]) {
+  } else if ([shapeLayer isKindOfClass:[MDCLegacyInkLayerBackgroundRipple class]]) {
     [self.backgroundRipples removeObject:(MDCLegacyInkLayerBackgroundRipple *)shapeLayer];
   }
+
+  if (self.isAnimating && self.foregroundRipples.count == 0 && self.backgroundRipples.count == 0) {
```

Comment:
Same as the `isAnimating` state above, we have to prevent sending the callback to early if not all (foreground and background) animations did finish.